### ISSUE TITLE
Kubernetes Ingress path should be configurable

### ISSUE THREAD

**Issue by ThoSap** [quarkusio/quarkus#26747]
### Describe the bug

Currently, the `quarkus-kubernetes` extension (2.10.2.Final) does not honor the value of Quarkus property `quarkus.http.root-path`, so the Kubernetes ingress path is always the root `/`.

### Expected behavior

The Kubernetes ingress path should equal property `quarkus.http.root-path`.

**application.properties**
```
quarkus.http.root-path=/newroot
```
**kubernetes.yml**
```yaml
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    app.quarkus.io/build-timestamp: 2022-07-14 - 15:06:39 +0000
  labels:
    app.kubernetes.io/name: quarkus-app
    app.kubernetes.io/version: 1.0.0
  name: quarkus-app
  namespace: my-namespace
spec:
  rules:
    - host: myhost.example.com
      http:
        paths:
          - backend:
              service:
                name: quarkus-app
                port:
                  name: http
            path: /newroot
            pathType: Prefix
```

### Actual behavior

**application.properties**
```
quarkus.http.root-path=/newroot
```
**kubernetes.yml**
```yaml
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    app.quarkus.io/build-timestamp: 2022-07-14 - 15:06:39 +0000
  labels:
    app.kubernetes.io/name: quarkus-app
    app.kubernetes.io/version: 1.0.0
  name: quarkus-app
  namespace: my-namespace
spec:
  rules:
    - host: myhost.example.com
      http:
        paths:
          - backend:
              service:
                name: quarkus-app
                port:
                  name: http
            path: /
            pathType: Prefix
```

### How to Reproduce?

https://code.quarkus.io/?j=17&e=kubernetes&e=resteasy-reactive&e=resteasy-reactive-jackson&extension-search=origin:platform%20resteasy

**application.properties**
```
quarkus.http.root-path=/newroot
```

### Output of `uname -a` or `ver`

Linux myhost 4.18.0-372.13.1.0.1.el8_6.x86_64 #1 SMP Fri Jul 1 15:06:54 PDT 2022 x86_64 x86_64 x86_64 GNU/Linux

### Output of `java -version`

OpenJDK 64-Bit Server VM Temurin-17.0.3+7 (build 17.0.3+7, mixed mode, sharing)

### GraalVM version (if different from Java)

OpenJDK 64-Bit Server VM GraalVM CE 22.1.0 (build 17.0.3+7-jvmci-22.1-b06, mixed mode, sharing)

### Quarkus version or git rev

2.10.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 7.4.2

**Comment by geoand** [comment]
Sounds reasonable, WDYT @iocanel ?

**Comment by ThoSap** [comment]
Hmmm, maybe it is not as easy as I thought.
What about the case when somebody sets the properties like this:
https://quarkus.io/blog/path-resolution-in-quarkus/#move-the-non-application-root-path-q
```
quarkus.http.root-path=/newroot/v1
quarkus.http.non-application-root-path=/newroot
```
Then for example `/newroot/swagger-ui/` would not work.

WDYT @geoand ?

**Comment by geoand** [comment]
Interesting point

**Comment by ThoSap** [comment]
Maybe the best solution would be to let the user set the HTTP ingress path with a new property **`quarkus.kubernetes.ingress.path`**?

My use case is the following. 
I have, for example, the domain `api.prod.example.com` and different Quarkus microservices are served using different K8s ingress paths:
https://api.prod.example.com/api/fights
https://api.prod.example.com/api/heroes
https://api.prod.example.com/api/villains

Without such property the generated Kubernetes YAML/JSON need to be post-processed (for example with sed) to replace K8s ingress path defintion:
`spec.rules[*].http.paths[*].path=/`
with
`spec.rules[*].http.paths[*].path=/api/fights`

**Comment by geoand** [comment]
Let's see what @iocanel thinks

**Comment by Sgitario** [comment]
> Maybe the best solution would be to let the user set the HTTP ingress path with a new optional ingress property **`quarkus.kubernetes.ingress.path`**?
> 
> My use case is the following. I have, for example, the domain `api.prod.example.com` and different Quarkus microservices are served using different K8s ingress paths: https://api.prod.example.com/api/fights https://api.prod.example.com/api/heroes https://api.prod.example.com/api/villains
> 
> Without such property the generated Kubernetes YAML/JSON need to be post-processed (for example with sed) to replace K8s ingress path defintion: `spec.rules[*].http.paths[*].path=/` with `spec.rules[*].http.paths[*].path=/api/fights`

You can already set the ingress host property: https://github.com/quarkusio/quarkus/blob/f4238ad15d1a3d73a83b4c2bf8945edf72ec4db6/extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/IngressConfig.java#L22
This change was added in https://github.com/quarkusio/quarkus/commit/788b5bec7edb5bb347d7bd75f9b0dd939d6686c4. 

Let me know if this works for you and if we can close the issue now.

**Comment by ThoSap** [comment]
Ahh cool, thanks! So the issue regarding secured ingress was fixed in Quarkus `2.12.0.CR1`, nice 😏

The PR/commit you mentioned fixes another workaround I made with a post-process step using `sed` for a secured ingress, but _does not fix this issue_ regarding the path explained here:
https://github.com/quarkusio/quarkus/issues/26747#issuecomment-1185267170

**Comment by Sgitario** [comment]
> @Sgitario @iocanel is this still an issue?

Yes, the only workaround is to provide a custom Ingress resource in `src/main/kubernetes/kubernetes.yml` with this definition. 
I've reported this issue in Dekorate to be fixed there first: https://github.com/dekorateio/dekorate/issues/1083

**Comment by Sgitario** [comment]
Quarkus main has been updated using the latest version of Dekorate which includes https://github.com/dekorateio/dekorate/issues/1083. So, I will be working on enabling this configuration in Quarkus asap.

**Comment by ThoSap** [comment]
Is this going to be merged in Quarkus `2.14.0.Final`?

**Comment by geoand** [comment]
That's the plan
